src/LandIce/evaluators/LandIce_BasalFrictionCoefficient_Def.hpp

```diff
@@ -242,115 +303,130 @@ KOKKOS_INLINE_FUNCTION
+  if(beta_type != GIVEN_CONSTANT) {
+    for (unsigned int ipt=0; ipt<dim; ++ipt) {
+
+      switch (effectivePressure_type) {
+      case EFFECTIVE_PRESSURE_TYPE::FIELD:
+        if(zero_N_on_floating_at_nodes) {
+          Nval = 0.0;
+          if(nodal) {
+            if (rho_i*thickness_field(cell,ipt)+rho_w*bed_topo_field(cell,ipt) > 0)
+              Nval = N(cell,ipt);
+          } else {
+            for (unsigned int node=0; node<numNodes; ++node)
+              if (rho_i*thickness_field(cell,node)+rho_w*bed_topo_field(cell,node) > 0)
+                Nval += N(cell,node)*BF(cell,node,ipt);
           }
-        } else {
-          for (unsigned int ipt=0; ipt<dim; ++ipt) {
-            ScalarT Nval = N(cell,ipt);
-            if (is_side_equation) Nval = KU::max(Nval, 0.0);
-            ScalarT q = u_norm(cell,ipt) / ( u_norm(cell,ipt) + lambdaField(cell,ipt)*ice_softness(cell)*std::pow(Nval,n) );
-            beta(cell,ipt) = mu * Nval * std::pow( q, power) / u_norm(cell,ipt);
+        } else
+          Nval = N(cell,ipt);
+        break;
+      case EFFECTIVE_PRESSURE_TYPE::ONE:
+        if(zero_N_on_floating_at_nodes) {
+          Nval = 0;
+          if(nodal) {
+            if(rho_i*thickness_field(cell,ipt)+rho_w*bed_topo_field(cell,ipt) > 0)
+              Nval =  1.0;
+          } else {
+            for (unsigned int node=0; node<numNodes; ++node)
+              if (rho_i*thickness_field(cell,node)+rho_w*bed_topo_field(cell,node) > 0)
+                Nval += BF(cell,node,ipt);
           }
         }
-      } else {
-        if (distributedMu) {
-          for (unsigned int ipt=0; ipt<dim; ++ipt) {
-            ScalarT Nval = N(cell,ipt);
-            if (is_side_equation) Nval = KU::max(Nval, 0.0);
-            ScalarT q = u_norm(cell,ipt) / ( u_norm(cell,ipt) + lambda*ice_softness(cell)*std::pow(Nval,n) );
-            beta(cell,ipt) = muCoulombField(cell,ipt) * Nval * std::pow( q, power) / u_norm(cell,ipt);
-          }
-        } else {
-          for (unsigned int ipt=0; ipt<dim; ++ipt) {
-            ScalarT Nval = N(cell,ipt);
-            if (is_side_equation) Nval = KU::max(Nval, 0.0);
-            ScalarT q = u_norm(cell,ipt) / ( u_norm(cell,ipt) + lambda*ice_softness(cell)*std::pow(Nval,n) );
-            beta(cell,ipt) = mu * Nval * std::pow( q, power) / u_norm(cell,ipt);
+        break;
+      case EFFECTIVE_PRESSURE_TYPE::HYDROSTATIC:
+        if(nodal)
+          Nval = g*KU::max(rho_i*thickness_field(cell,ipt)+KU::min(rho_w*bed_topo_field(cell,ipt),0.0),0.0);
+        else {
+          MeshScalarT thickness(0), bed_topo(0);
+          for (unsigned int node=0; node<numNodes; ++node) {
+            thickness += thickness_field(cell,node)*BF(cell,node,ipt);
+            bed_topo += bed_topo_field(cell,node)*BF(cell,node,ipt);
           }
+          Nval = g*KU::max(rho_i*thickness+KU::min(rho_w*bed_topo,0.0),0.0);
+        }
+        break;
+      case EFFECTIVE_PRESSURE_TYPE::HYDROSTATIC_AT_NODES:
```

Comment:
Why computing N at nodes and then interpolating at qp's? Why not computing N directly at the qp's? It seems the latter is more accurate in general.

Reply:
I'd like that if you compute the hydrostatic effective pressure in MPAS, on nodes, and then pass it to Albany, you get the same result as computing that with Albany using `HYDROSTATIC_AT_NODES`. Also, if N is computed at quadrature points, it will be a nonsmooth field at the GL, which makes it hard to be integrated.